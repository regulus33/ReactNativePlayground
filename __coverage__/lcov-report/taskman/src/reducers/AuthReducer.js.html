<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for taskman/src/reducers/AuthReducer.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">taskman/src/reducers</a> AuthReducer.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.57% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/12</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {
  EMAIL_CHANGED,
  PASSWORD_CHANGED,
  LOGIN_USER_SUCCESS,
  LOGIN_USER_FAIL,
  LOGIN_USER
} from '../actions/types';
&nbsp;
const INITIAL_STATE = {
  email: '',
  password: '',
  user: null,
  error: '',
  loading: false
}
//remember, var setting in parameter definition is a null handler kind of thing.
export default (state = INITIAL_STATE, action) =&gt; {
  switch (action.type) {
    //if the action's type is EMAIL_CHANGED which is imported from above and is literally evaluating to a string literal
<span class="branch-0 cbranch-no" title="branch not covered" >    case EMAIL_CHANGED:</span>
<span class="cstat-no" title="statement not covered" >      state.email = action.payload;</span>
<span class="cstat-no" title="statement not covered" >      return {...state, email: action.payload} ;</span>
<span class="branch-1 cbranch-no" title="branch not covered" >    case PASSWORD_CHANGED:</span>
<span class="cstat-no" title="statement not covered" >      state.password = action.payload;</span>
<span class="cstat-no" title="statement not covered" >      return {...state, password: action.payload} ;</span>
<span class="branch-2 cbranch-no" title="branch not covered" >    case LOGIN_USER:</span>
<span class="cstat-no" title="statement not covered" >      return {...state, loading: true, error: ''}</span>
<span class="branch-3 cbranch-no" title="branch not covered" >    case LOGIN_USER_SUCCESS:</span>
<span class="cstat-no" title="statement not covered" >      console.log("Success")</span>
<span class="cstat-no" title="statement not covered" >      return {...state, ...INITIAL_STATE, user: action.payload}</span>
<span class="branch-4 cbranch-no" title="branch not covered" >    case LOGIN_USER_FAIL:</span>
<span class="cstat-no" title="statement not covered" >      console.log("Fail")</span>
<span class="cstat-no" title="statement not covered" >      return {...state, error: 'Authentication Failed.', loading: false}</span>
    default:
      return state;
  }
};</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Dec 18 2017 15:39:56 GMT-0500 (EST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
