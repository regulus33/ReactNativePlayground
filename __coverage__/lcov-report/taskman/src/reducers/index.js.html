<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for taskman/src/reducers/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">taskman/src/reducers</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { combineReducers } from 'redux';
import AuthReducer from './AuthReducer';
import TaskFormReducer from './TaskFormReducer'
import TasksReducer from './TasksReducer'
&nbsp;
export default combineReducers({
    // auth piece of state is produced by the Auth reducer
    auth: AuthReducer,
    taskForm: TaskFormReducer,
    tasks: TasksReducer
});
&nbsp;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Dec 18 2017 15:39:56 GMT-0500 (EST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
