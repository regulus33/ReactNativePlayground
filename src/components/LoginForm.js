import React, { Component } from 'react';
import { Card, CardSection, Button, Input } from './common';
import { emailChanged, passwordChanged } from '../actions';
import { connect } from 'react-redux';

class LoginForm extends Component {

  onEmailChange(text) {
    this.props.passwordChanged(text)
  }

  onPassWordChange(text) {
    this.props.passwordChanged(text)
  }

  render() {
    return(
       <Card>
        <CardSection>
          <Input
            placeholder="user@gmail.com"
            label="Email"
            value={this.props.email}
            onChangeText={this.onEmailChange.bind(this)}
          />
        </CardSection>

        <CardSection>
          <Input
            placeholder="Password"
            secureTextEntry
            label="Password"
            onChangeText={this.onPassWordChange.bind(this)}
          />
        </CardSection>

        <CardSection>
          <Button>
            Login
          </Button>
        </CardSection>
      </Card>
    );
  }
}

const styles={
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red'
  }
}

const mapStateToProps = state => {
  return {
    email: state.auth.email
  };
};

export default LoginForm;