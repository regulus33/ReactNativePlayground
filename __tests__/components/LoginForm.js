import 'react-native';
import React from 'react';
import LoginForm from '../../src/components/LoginForm';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import reducers from '../../src/reducers';
import ReduxThunk from 'redux-thunk';
import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';

import configureStore from 'redux-mock-store';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

// imported as a connected component!

const middlewares = []; // you can mock any middlewares here if necessary
// const mockStore = configureStore(middlewares);

console.log(configureStore(reducers, {}, applyMiddleware(ReduxThunk)))
const mockStore = configureStore(reducers, {}, applyMiddleware(ReduxThunk))



const initial = {
  auth: {
    email: '',
    password: '',
    user: null,
    error: '',
    loading: false
  }
}

describe('Testing LoginForm', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <LoginForm/>,
      { context: { store: mockStore(initial) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
