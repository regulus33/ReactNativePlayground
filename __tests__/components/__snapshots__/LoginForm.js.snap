// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CameraSettings renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    email=""
    emailChanged={[Function]}
    error=""
    loading={false}
    loginUser={[Function]}
    password=""
    passwordChanged={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardSection>
          <Input
            label="Email"
            onChangeText={[Function]}
            placeholder="email@gmail.com"
            value=""
          />
        </CardSection>,
        <CardSection>
          <Input
            label="Password"
            onChangeText={[Function]}
            placeholder="password"
            secureTextEntry={true}
            value=""
          />
        </CardSection>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "red",
              "fontSize": 20,
            }
          }
        >
          
        </Text>,
        <CardSection>
          <Button
            onPress={[Function]}
          >
            Login
          </Button>
        </CardSection>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Input
            label="Email"
            onChangeText={[Function]}
            placeholder="email@gmail.com"
            value=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Email",
            "onChangeText": [Function],
            "placeholder": "email@gmail.com",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Input
            label="Password"
            onChangeText={[Function]}
            placeholder="password"
            secureTextEntry={true}
            value=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Password",
            "onChangeText": [Function],
            "placeholder": "password",
            "secureTextEntry": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "",
          "ellipsizeMode": "tail",
          "style": Object {
            "alignSelf": "center",
            "color": "red",
            "fontSize": 20,
          },
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Button
            onPress={[Function]}
          >
            Login
          </Button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Login",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardSection>
            <Input
              label="Email"
              onChangeText={[Function]}
              placeholder="email@gmail.com"
              value=""
            />
          </CardSection>,
          <CardSection>
            <Input
              label="Password"
              onChangeText={[Function]}
              placeholder="password"
              secureTextEntry={true}
              value=""
            />
          </CardSection>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "red",
                "fontSize": 20,
              }
            }
          >
            
          </Text>,
          <CardSection>
            <Button
              onPress={[Function]}
            >
              Login
            </Button>
          </CardSection>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              label="Email"
              onChangeText={[Function]}
              placeholder="email@gmail.com"
              value=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "Email",
              "onChangeText": [Function],
              "placeholder": "email@gmail.com",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              label="Password"
              onChangeText={[Function]}
              placeholder="password"
              secureTextEntry={true}
              value=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "Password",
              "onChangeText": [Function],
              "placeholder": "password",
              "secureTextEntry": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "",
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "color": "red",
              "fontSize": 20,
            },
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              onPress={[Function]}
            >
              Login
            </Button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Login",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "Login",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
